test/test_component.py

```diff
@@ -47,6 +47,15 @@ def test_system_building(self):
         comparison_file = os.path.join(test_dir, 'test_files', 'component_construction.nt')
         assert filecmp.cmp(tmp_out, comparison_file), f'Converted file {tmp_out} is not identical'
 
+        hlc_doc = sbol3.Document()
```

Comment:
Three changes to request:
1) These should be in their own test function (e.g. `def test_high_level_constructors(self)`, not in `test_system_building`
2) The test should check that the _contents_ of the constructed components, not just that they validate. You can do this by writing a copy of the test document, checking it by hand for correctness, then comparing the results of your builder calls to make sure the document you produce is the same as the hand-checked test document. This is what is done with `test_system_building`
3) The test should exercise all of the new functions, not just a subset.

The first change will make it easier to debug failures.
The second and third changes are important to make sure that your functions are producing the desired result.

Reply:
@jakebeal to Implement the proposed test I though in something like this for every constructor:

```
dna_identity = 'dna_component_with_sequence'
dna_sequence = 'ttt'
test_description = 'test'
hl_dna_comp, hl_dna_seq = dna_component_with_sequence(dna_identity, dna_sequence, description=test_description)
dna_seq = sbol3.Sequence(f'{dna_identity}_seq',
                        elements=dna_sequence,
                        encoding=sbol3.IUPAC_DNA_ENCODING)

dna_comp = sbol3.Component(dna_identity, sbol3.SBO_DNA,
                        sequences=[dna_seq],
                        description=test_description)
if dna_comp.name == None:
    dna_comp.name = dna_identity
assert hl_dna_comp == dna_comp , f'Constructor Error: {dna_identity}'
```
Doing this I notice that the components are not equal, is this expected because they are indeed different components with the same information? How can I compare them for the test?

Reply:
That's right: they are not equal because they are different objects. The sbol_diff package has functions for comparing objects, however, which you may find useful for this purpose.

Reply:
@jakebeal I updated the test as you required, please take a look at the `test_high_level_constructors` implementation using `doc_diff` which is the only tool that I found useful for this. It would be more useful for me to have a `toplvl_diff` or something like that, that sounds useful for other workflows?

Reply:
doc_diff looks like it's working for you here. If you think toplvl_diff would be useful, please open an issue for it!

Reply:
This request is on issue #94 now